src/Workspaces/Core/Portable/NamingStyles/NamingStyle.cs

```diff
@@ -70,11 +70,11 @@ private IEnumerable<string> ApplyCapitalization(IEnumerable<string> words)
             switch (CapitalizationScheme)
             {
                 case Capitalization.PascalCase:
-                    return words.Select(CapitalizeFirstLetter);
+                    return words.Select(s_capitalizeFirstLetter);
```

Comment:
These were intentionally not changed when i did the perf pass for Naming.  The expensive hotpaths are in analysis.  This was a codepath that was only hit when you needed to actually fix a name.  And as such having an allocation here was fine.  After all, we're already going to get another allocation from the .Select itself.   

THere are codepaths where being alloc-free (or close thereof) is really important.  I'd prefer we spend our efforts there.